Test/2022 UI api plans run e2e

kkibbey1996 opened the pull request:
## Issue

https://gravitee.atlassian.net/browse/APIM-2022 

## Description

Plans page tests for APIs in APIM. 

Splitting out my General and Plan commits, with updated code and faker details. 

I will squash commits etc. after review comments & changes. 

## Additional context

<!-- Add any other context about the PR here -->
<!-- It can be links to other PRs or docs or drawing -->
<!-- Or reproduction steps in case of bug fix -->
<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-audrnpyuji.chromatic.com)
<!-- Storybook placeholder end -->

phiz71 commented:
@kkibbey1996 I don't think you made 113 commits 😅. Could you please rebase your branch to keep only the commits you've done for this PR? 🙏🏻

kkibbey1996 committed:
test(ui): api plans tests

chore(deps): update helm release elasticsearch to v19.10.6

fix: e2e tests for API ownership transfer

fix(e2e-mapi-v2): reorder resource clean-up in afterAll

fix: align Alert Engine CTA's with other OS

ci: remove git branch creation and PR process in Helm release

Revised the process of releasing Helm charts so that it no longer creates a unique release branch or a PR. Instead, changes are committed and pushed directly on the gh-pages branch. This simplifies the workflow, accelerates the release process, and reduces the need for manual PR reviews and merges.

(cherry picked from commit d371bbcde4142250df9e38fa3be560ab3072ad38)

fix: display only accessible menu items

https://gravitee.atlassian.net/browse/APIM-2395
(cherry picked from commit b0d6fc0912cfa818866c4a0507da6b4303ef0fcb)

fix: fix some screen in readonly mode

(cherry picked from commit b056c5791aed42f29f69236d928b7d2e079bc1e3)

julian-robotham-gravitee commented on gravitee-apim-e2e/ui-test/integration/apim/ui/apis/ui-api-plans.spec.ts:
```diff
@@ -0,0 +1,265 @@
+    cy.contains(`${planName}-APIKey`).should('be.visible');
+    cy.getByDataTestId('api_plans_closeplan_button').first().click();
+    cy.get(`[placeholder="${planName}-APIKey"]`).type(`${planName}-APIKey`);
+    // cy.contains('Yes, delete this plan').click()
```
Is this comment still needed?

julian-robotham-gravitee commented on gravitee-apim-console-webui/src/management/api/general/plans/list/api-portal-plan-list.component.html:
```diff
@@ -107,27 +108,57 @@
+              mat-icon-button
+              aria-label="Edit the plan"
+              matTooltip="Edit the plan"
+              data-testid="api_plans_editplan_button"
```
For the reference name and value I would of suggested  `data-test-id="api_plans_edit_plan_button"`
but this is probably a minor point and shouldn't block the PR

Okhelifi replied:
At the beginning of the cypress tests developments the `data-testid` usage came from here https://docs.cypress.io/guides/references/best-practices#How-It-Works I think 🤷🏼 Maybe I'm wrong

kkibbey1996 committed:
Revert "test(ui): api plans tests"

This reverts commit 45eb33163099a222c9acfb0dc2bc9dbdcfe71b02.

Okhelifi reviewed (changes requested):
You have an issue with your last rebase I think.
